packages/ui-components/src/components/DateTimePicker/DateTimePicker.component.tsx

```diff
@@ -0,0 +1,700 @@
+  jn-py-4
+`
+
+const inputDefaultBorderStyles = `
+  jn-border-theme-textinput-default
+`
+
+const inputInvalidStyles = `
+  jn-border-theme-error
+`
+
+const inputValidStyles = `
+  jn-border-theme-success
+`
+
+const labelStyles = `
+  peer-autofill:jn-text-theme-textinput-autofill-label
+  jn-pointer-events-none
+  jn-top-2
+  jn-left-[0.9375rem]
+`
+
+const iconContainerStyles = `
+  jn-absolute
+  jn-inline-flex
+  jn-top-2
+  jn-right-[2.75rem]
+  jn-gap-1.5
+`
+
+interface SelectedDate {
+  selectedDate?: Date[]
+  selectedDateStr?: string
+}
+
+/** A all-purpose date and time picker component. Highly configurable, based on Flatpickr. */
+
+export const DateTimePicker: React.FC<DateTimePickerProps> = ({
+  allowInput = false,
+  allowInvalidPreload = false,
+  ariaDateFormat = "F j, Y",
+  className = "",
+  conjunction = ", ",
+  dateFormat = undefined,
+  defaultHour = 12,
+  defaultMinute = 0,
+  defaultDate = null,
+  defaultValue = "",
+  disable = [],
+  disabled = false,
+  enableSeconds = false,
+  enableTime = false,
+  errortext = "",
+  helptext = "",
+  hourIncrement = 1,
+  id = "",
+  invalid = false,
+  label = "",
+  locale = null,
+  maxDate = null,
+  minDate = null,
+  minuteIncrement = 1,
+  mode = "single",
+  monthSelectorType = "static",
+  name = "",
+  noCalendar = false,
+  onBlur = undefined,
+  onChange = undefined,
+  onClear = undefined,
+  onClose = undefined,
+  onFocus = undefined,
+  onMonthChange = undefined,
+  onOpen = undefined,
+  onReady = undefined,
+  onYearChange = undefined,
+  placeholder = "",
+  required = false,
+  shorthandCurrentMonth = false,
+  showMonths = 1,
+  successtext = "",
+  time_24hr = false,
+  valid = false,
+  value = "",
+  weekNumbers = false,
+  width = "full",
+  wrapperClassName = "",
+  ...props
+}) => {
+  // always generate auto-id string using the useId hook to avoid "more hooks than in previous render" error when removing custom id:
+  const autoId = "juno-datetimepicker-" + useId()
+  const theId = id && id.length ? id : autoId
+
+  const fpRef = useRef<HTMLInputElement | null>(null) // the DOM node flatpickr instance will be bound to
+  const flatpickrInstanceRef = useRef<flatpickr.Instance | null>(null) // The actual flatpickr instance
+  const calendarTargetRef = useRef<HTMLDivElement | null>(null) // The DOM node the flatpickr calendar should be rendered to
+
+  const [theDate, setTheDate] = useState<SelectedDate>({})
+  // variables starting with underscore are not linted
+  const [_hasFocus, setHasFocus] = useState<boolean>(false)
+  const [isOpen, setIsOpen] = useState<boolean>(false)
+  const [isInvalid, setIsInvalid] = useState<boolean>(false)
+  const [isValid, setIsValid] = useState<boolean>(false)
+
+  const hasLength = (node: React.ReactNode) => (typeof node === "string" || Array.isArray(node)) && node.length
+
+  const updateFlatpickrInstance = (newKeys: flatpickr.Instance | null) => {
+    if (flatpickrInstanceRef.current === null) {
```

Comment:
**(Question)**

That could be a potential breaking change or fix since you change the logic that was before in this method to use object.keys, please provide explanation for that.

Reply:
I had some serious troubles solution with object.keys to work in typescript, while it seemed to just do what current script does.
Could you describe the difference, so that I could know and try to make correction?

Reply:
I left the new one, because I don't see functional difference. However there is also this equivalent

  const updateFlatpickrInstance = (newKeys: flatpickr.Instance | null) =>
    (flatpickrInstanceRef.current = {
      ...flatpickrInstanceRef.current,
      ...Object.keys(newKeys!).reduce((a, key) => {
        const k = key as keyof flatpickr.Instance
        const newVal = newKeys![key as keyof flatpickr.Instance]
        a[k] = newVal as never
        return a
      }, {} as flatpickr.Instance),
    } as flatpickr.Instance)
    
    what I'm unhappy about here is amount of casting and casting to 'never' - not sure if this would even pass verification pipeline.